packages/near-membrane-base/src/membrane.ts

```diff
@@ -1644,4 +1644,17 @@ export function createMembraneMarshall() {
     };
 }
 
-export const marshallSourceTextInStrictMode = `(function(){'use strict';return (${createMembraneMarshall.toString()})})()`;
+const ErrorCtor = Error;
+const marshallSourceTextInStrictMode = `(function(){'use strict';return (${createMembraneMarshall.toString()})})()`;
+export function createConnectorForGlobalObject(globalObject: typeof globalThis) {
+    if (!globalObject) {
+        throw new ErrorCtor('Missing global object');
+    }
+    if (typeof globalObject.eval === 'undefined') {
+        throw new ErrorCtor('Missing global object eval');
+    }
+    // The result of this eval will be a function that returns a function.
+    // The hooks connector is the last returned function, so we invoke the
+    // result of the eval operation and return that result.
+    return globalObject.eval(marshallSourceTextInStrictMode)();
```

Comment:
Can we pluck and reuse just the `eval` reference and not access it repeatedly as `globalObject.eval`. Could user code modify the `.eval` reference`?

Reply:
I don't think what I'm doing here is much different than what it was doing before 🤷  This global object is always created right before this function is called.

Reply:
Those are valid concerns, but as @rwaldron mentioned, most of the time, the redGlobalThis is fresh window from same domain iframe... now... that's not the case of ShadowRealm. So, this change seems incompatible with ShadowRealm. In the shadowRealm instance you do not have access to the `globalThis` from outside, and therefore you won't be able to evaluate init on it. This seems fatal.

Reply:
That's true, but centralizing as we're doing here will allow us to replace a single function's definition from: 

```
export function createConnectorForGlobalObject(globalObject: typeof globalThis) {
    if (!globalObject) {
        throw new TypeErrorCtor('Missing global object');
    }
    if (typeof globalObject.eval === 'undefined') {
        throw new TypeErrorCtor('Missing global object eval');
    }
    // The result of this eval will be a function that returns a function.
    // The hooks connector is the last returned function, so we invoke the
    // result of the eval operation and return that result.
    return globalObject.eval(marshallSourceTextInStrictMode)();
}
```

to: 

```
export function createConnectorForRealm(shadowRealm) {
    if (!shadowRealm) {
        throw new TypeErrorCtor('Missing shadow realm');
    }
    // The result of this eval will be a function that returns a function.
    // The hooks connector is the last returned function, so we invoke the
    // result of the eval operation and return that result.
    return shadowRealm.evaluate(marshallSourceTextInStrictMode)();
}
```

Reply:
Even more to the point...

Putting this operation in a function now makes migration to ShadowRealm nicer if that step is consolidated into a function that replaces this one: 

```ts
export function createConnectorForRealm() {
    const r = new ShadowRealm();
    // The result of this eval will be a function that returns a function.
    // The hooks connector is the last returned function, so we invoke the
    // result of the eval operation and return that result.
    return r.evaluate(marshallSourceTextInStrictMode)();
}
```